common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java

```diff
@@ -123,6 +139,84 @@ private static void findQuestForTracking(ContainerContent container, QuestInfo q
         }
     }
 
+    public static void scanDialogueHistory() {
+        ScriptedContainerQuery.QueryBuilder queryBuilder = ScriptedContainerQuery.builder(
+                        "Quest Book Dialogue History Query")
+                .onError(msg -> WynntilsMod.warn("Problem getting dialogue history in Quest Book: " + msg))
+                .useItemInHotbar(InventoryUtils.QUEST_BOOK_SLOT_NUM)
+                .matchTitle(getQuestBookTitle(1))
+                .processContainer((c) -> {
+                    ItemStack dialogueHistoryItem = c.items().get(0);
+
+                    if (!ComponentUtils.getCoded(dialogueHistoryItem.getHoverName())
+                            .equals("§bDialogue History")) return;
+
+                    for (String line : ItemUtils.getLore(dialogueHistoryItem)) {
+                        Matcher matcher = DIALOGUE_HISTORY_PAGE_PATTERN.matcher(line);
+
+                        if (matcher.matches()) {
+                            int pageCount = Integer.parseInt(matcher.group(2));
+                            createActualDialogueHistoryQuery(pageCount);
+                            break;
+                        }
+                    }
+                });
+
+        queryBuilder.build().executeQuery();
+    }
+
+    private static void createActualDialogueHistoryQuery(int pageCount) {
+        ScriptedContainerQuery.QueryBuilder queryBuilder = ScriptedContainerQuery.builder(
+                        "Quest Book Dialogue History Query 2")
+                .onError(msg -> WynntilsMod.warn("Problem getting dialogue history (2) in Quest Book: " + msg))
+                .useItemInHotbar(InventoryUtils.QUEST_BOOK_SLOT_NUM)
+                .matchTitle(getQuestBookTitle(1))
+                .setWaitForMenuReopen(false)
+                .processContainer((c) -> {
+                    ItemStack dialogueHistoryItem = c.items().get(0);
+
+                    if (!ComponentUtils.getCoded(dialogueHistoryItem.getHoverName())
+                            .equals("§bDialogue History")) return;
+
+                    newDialogueHistory = new ArrayList<>();
+
+                    List<String> current = ItemUtils.getLore(dialogueHistoryItem).stream()
+                            .dropWhile(String::isBlank)
+                            .takeWhile(s -> !s.isBlank())
+                            .toList();
+
+                    newDialogueHistory.add(current);
+                });
+
+        for (int i = 2; i <= pageCount; i++) {
+            int page = i;
+            queryBuilder
+                    .clickOnSlot(0)
+                    .matchTitle(getQuestBookTitle(1))
+                    .setWaitForMenuReopen(false)
+                    .processContainer((c) -> {
+                        ItemStack dialogueHistoryItem = c.items().get(0);
+
+                        if (!ComponentUtils.getCoded(dialogueHistoryItem.getHoverName())
+                                .equals("§bDialogue History")) return;
```

Comment:
This check is only necessary once in `scanDialogueHistory` technically, so check could be reasonably removed. It is fine to keep, however.

Reply:
Yeah, it is just safer to do so.